aip/0194.md

```diff
@@ -0,0 +1,237 @@
+---
+aip:
+  id: 194
+  state: reviewing
+  created: 2019-08-16
+permalink: /194
+redirect_from:
+  - /0194
+---
+
+# Retrying for client libraries
+
+This document aims to lay out guidance about which requests to retry, and for
+which codes.
+
+Note: this document often uses the term "CRUD". This acronym stands for
+Create, Read (Get, List, etc.), Update, Delete.
+
+## Guidance
+
+Requests can be categorized as either unary or streaming. A unary request is
+a single request with a single response. A streaming request involves multiple
+pieces of data being sent.
+
+Unary requests can be further categorized as:
+
+- Retryable: requests in which repeated runs would not cause unintended state
```

Comment:
I think we need to split this: 1 section for defining things, 1 section for saying what should and should not be retried.